drivers/spi/spi_esp32_spim.c

```
@@ -214,6 +214,39 @@ static int spi_esp32_init_dma(const struct device *dev)
 	return 0;
 }
 
+static int spi_esp32_deinit_dma(const struct device *dev)
+{
+	const struct spi_esp32_config *cfg = dev->config;
+	struct spi_esp32_data *data = dev->data;
+
+	spi_hal_deinit(&data->hal);
+
+	data->hal_config.dma_in = NULL;
+	data->hal_config.dma_out = NULL;
+	data->hal_config.dma_enabled = false;
+	data->hal_config.tx_dma_chan = 0;
+	data->hal_config.rx_dma_chan = 0;
+	data->hal_config.dmadesc_n = 0;
+	data->hal_config.dmadesc_rx = NULL;
+	data->hal_config.dmadesc_tx = NULL;
+
+#ifdef SOC_GDMA_SUPPORTED
+	/* FIXME: need gdma_hal_deinit() */
```

Would it be better to let this fail during build- or runtime? I'm definitely unsure what the consequences are of this part missing, but a silent comment gets easily overlooked.

---

Fishing for suggestions from @sylvioalves or another esp32 maintainer.

---

@cfriedt, regarding GDMA I think you must call `gdma_ll_tx_disconnect_to_periph()` and `gdma_ll_rx_disconnect_to_periph()` inside your implementation of `spi_esp32_deinit_dma()`, but be careful regarding the possibility of calling `gdma_ll_enable_clock(data->hal_gdma.dev, false)` since current esp32 spi driver implementation is not taking in account that there is an esp32 Zephyr (G)DMA device driver ([issue 87147#issuecomment-2730731548](https://github.com/zephyrproject-rtos/zephyr/issues/87147#issuecomment-2730731548)).

---

Thanks @wmrsouza !